generator/src/main/java/com/scottlogic/deg/generator/Guice/DecisionTreeWalkerProvider.java

```diff
@@ -0,0 +1,35 @@
+package com.scottlogic.deg.generator.Guice;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.name.Named;
+import com.scottlogic.deg.generator.generation.GenerationConfigSource;
+import com.scottlogic.deg.generator.walker.DecisionTreeWalker;
+
+public class DecisionTreeWalkerProvider implements Provider<DecisionTreeWalker> {
+    private final DecisionTreeWalker reductiveDecisionTreeWalker;
+    private final DecisionTreeWalker cartesianProductDecisionTreeWalker;
+    private final GenerationConfigSource commandLine;
+
+    @Inject
+    public DecisionTreeWalkerProvider(
+        @Named("reductive") DecisionTreeWalker reductiveDecisionTreeWalker,
```

Comment:
To be agreed what is the best approach here. For now using `@Named("xxx")` is ok, but in phase 2 (or the future) we may want to change these.

Reply:
We should add the other walker types to this tho. Think it might be just `ROUTED`, but its available before this PR so it shouldn't be removed as part of the refactoring.

Reply:
`ROUTED` has now been added